### PR TITLE
write about the constants vs hashing

### PR DESCRIPTION
**Pull request by alexandernutz**
this sprang from this discussion:
https://discord.com/channels/795999272293236746/1174649009697193995

Jira ticket:https://certora.atlassian.net/browse/DOC-358
Link to generated documentation: https://certora-certora-prover-documentation--175.com.readthedocs.build/en/175/

### CODE REVIEW

File: docs/prover/approx/hashing.md [hunk 1]
```diff
@@ -33,33 +33,36 @@ inputs `x` and `y`
...
-We illustrate the consequences of our modeling on a few examples.
+We illustrate the implications of our modeling decisions using a few examples.
 
 ### Modeling does not account for individual values of the Keccak function
 
 The Keccak256-hash of the string `hello` is 
 `0x1c8aff950685c2ed4bc3174f3472287b56d9517b9c948127319a09a7a36deac8`.
-However, due to our modeling, Certora Prover cannot prove that fact. The rule 
-`hashOf17Eq` will show as "violated" since Certora Prover can pick a function 
+However, due to our modeling, the certora prover cannot prove that fact. the
```
Comment (yoav-rodeh) [inline docs/prover/approx/hashing.md:62]: shouldn't certora prover be capitalized?

**Review by yoav-rodeh — APPROVED** [review]
impressive doc!

File: docs/user-guide/glossary.md [hunk 2]
```diff
@@ -50,8 +61,13 @@ model
 example
 counterexample
   The terms "model", "example", and "counterexample" are used interchangeably.
-  They all refer to an assignment of values to all of the CVL variables and
-  contract storage.  See {ref}`rule-overview`.
+  In the context of a CVL rule, they refer to an assignment of values to all of 
+  the CVL variables and contract storage that either violates an `assert` statement 
+  or fulfills a `satisfy` statement. See {ref}`rule-overview`.
```
Comment (nd-certora) [inline docs/user-guide/glossary.md:66]: for satisfy we use the term "witness example"

File: docs/prover/approx/hashing.md [hunk 3]
```diff
@@ -117,8 +120,77 @@ contract C {
...
+`keccak(1, k)` equals 1000000, the map update would overwrite that storage
+slot, and thus the assertion in the rule `foo` would be violated.
+
+However, the Certora Prover will return "not violated" for this assertion, since
+it assumes that no hash ever collides with the constant 1000000, which occurs in
+the program.
+
+On the other hand, if we change the contract to leave `slotAddress`
+uninitialized, then Certora Prover will return a violation, since then it can
+choose the values such that `keccak(2, preImage)` == `slotAddress`.
```
Comment (nd-certora) [inline docs/prover/approx/hashing.md:183]: maybe give a link to a violation that one can see how it looks

**Review by nd-certora — APPROVED** [review]
Good, two comments on improvements we can make, not mandatory

File: docs/user-guide/glossary.md (hunk 2 above)
Comment (alexandernutz) [reply docs/user-guide/glossary.md:66]: ah, nice .. tweaking this entry and the `model`/`example/.. one

File: docs/prover/approx/hashing.md (hunk 3 above)
Comment (alexandernutz) [reply docs/prover/approx/hashing.md:183]: Huh, you mean to give a link to a run?
Didn't think about doing that in our docs, but yeah, why not -- they just need to be stable, but I guess for the forseeable future they are ..

**Pull request merged**